submissions/Roophee/pair-game/index.js

```diff
@@ -109,61 +128,61 @@ const addCardsOnPage = function () {
 
 addCardsOnPage();
 
-//  CLEAR ARRAYS AND OBJECTS, PICK NEW CARDS FOR THE GAME
+//  CLEAR OPERATION OBJECTS, PICK NEW CARDS FOR THE GAME
 const clearAllOperationObjectsAndCreateNewGame = function() {
-    if (pairs.length === maxQuantityOfPairsOnTheScreen) {
-        const playAgain = window.confirm('Congratulations. You have matched all pairs and won the game. Want to play again?');
-                if (playAgain) {
-                    cardPlace.innerHTML = "";
-                    for (const prop of Object.keys(imagesCountForTwo)) {
-                        delete imagesCountForTwo[prop];
-                        };
-                    indexInUse.length = 0;
-                    prePair.length = 0;
-                    pairs.length = 0;
-                    cardsForPlay();
-                    addCardsOnPage();
-                }
-            }
+    const playAgain = window.confirm('Congratulations. You have matched all pairs and won the game. Want to play again?');
+        if (playAgain) {
+            cardPlace.innerHTML = "";
+            prePair.length = 0;
+            pairs.length = 0;
+            addCardsOnPage();
+        }
 }
 
 // CHECK PLAYER COLLECTED ALL PAIRS OR NOT
 const checkGameIsOverOrNot = function () {
-    setTimeout(function () {
-            clearAllOperationObjectsAndCreateNewGame();
-    }, 2100)
+    if (pairs.length === maxQuantityOfPairsOnTheScreen){
+        setTimeout(function () {
+                clearAllOperationObjectsAndCreateNewGame();
+        }, 2100)
+    }
 }
 
 // ADD HIDE CLASS ON CARD-HOLDER ITEM IF TWO CARD ARE PAIR
-let hidePair = function(){
-    if (prePair[0] && prePair[1]){
-    prePair[0][1].classList.add('hide');
-    prePair[1][1].classList.add('hide');
-    prePair.splice(0, 2);
+let hidePair = function(currentPair){
+    if (currentPair[0] && currentPair[1]){
+        currentPair[0][1].classList.add('hide');
```

Comment:
Try to avoid variable calls via indexes `currentPair[0][1]` Not readable. I as person who do not work with this code everyday can't understand and remember what data stored here.